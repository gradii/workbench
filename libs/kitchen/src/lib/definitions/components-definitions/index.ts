export * from './tabs/tabs-definition';
export * from './tabs/tabs-definition.module';
export * from './sidebar/sidebar-definition';
export * from './sidebar/sidebar-definition.module';
export * from './multiple-axis-chart/multiple-axis-chart-definition.module';
export * from './multiple-axis-chart/data-validator';
export * from './multiple-axis-chart/multiple-axis-chart-definition';
export * from './multiple-axis-chart/multiple-axis-chart.component';
export * from './multiple-bar-chart/multiple-bar-chart-definition.module';
// export * from './multiple-bar-chart/data-validator';
export * from './multiple-bar-chart/multiple-bar-chart.component';
export * from './multiple-bar-chart/multiple-bar-chart-definition';
export * from './iframe/iframe-definition.module';
export * from './iframe/iframe-definition';
export * from './calendar/calendar-definition.module';
export * from './calendar/calendar-definition';
export * from './radio/radio-definition';
export * from './radio/radio-definition.module';
export * from './card/card-definition.module';
export * from './card/card-definition';
export * from './input/input-definition.module';
export * from './input/input-definition';
export * from './accordion/accordion-definition';
export * from './accordion/accordion-definition.module';
// export * from './bubble-map/data-validator';
export * from './heading/heading-definition.module';
export * from './heading/heading-definition';
export * from './progress-bar/progress-bar-definition.module';
export * from './progress-bar/progress-bar-definition';
export * from './datepicker/datepicker-definition';
export * from './datepicker/datepicker-definition.module';
export * from './sidebar-toggle/sidebar-toggle-definition.module';
export * from './pie-chart/pie-chart-definition';
// export * from './pie-chart/data-validator';
export * from './pie-chart/pie-chart-definition.module';
export * from './pie-chart/pie-chart.component';
export * from './checkbox/checkbox-definition.module';
export * from './checkbox/checkbox-definition';
// export * from './bar-chart/data-validator';
export * from './stepper/stepper-definition';
export * from './stepper/stepper-definition.module';
export {
  ImageDefinitionModule
} from './image/image-definition.module';
export * from './image/image-definition';
export * from './smart-table/smart-table-triggers';
// export * from './smart-table/data-validator';
export * from './smart-table/smart-table-definition.module';
export * from './smart-table/smart-table-definition';
export * from './button-link/button-link-definition.module';
export * from './button-link/button-link-definition';
export * from './map/map-definition';
export * from './map/map.component';
export * from './map/map-definition.module';
export * from './button/button-definition';
export * from './button/button-definition.module';
export * from './link/link-definition';
export * from './link/link-definition.module';
export * from './table/table-definition';
export * from './table/table-definition.module';
export * from './list/list-definition.module';
export * from './list/list-definition';
export {
  DividerDefinitionModule
} from './divider/divider-definition.module';
export * from './divider/divider-definition';
export * from './menu/menu-definition';
export * from './menu/menu-definition.module';
// export * from './line-chart/data-validator';
export * from './text/text-definition.module';
export * from './text/text-definition';
export * from './select/select-definition';
export * from './select/select-definition.module';
export * from './icon/icon-definition';
export * from './icon/icon-definition.module';
export * from './space/space-definition';
export * from './space/appear-animation';
export * from './space/space-definition.module';
export * from './doughnut-chart/doughnut-chart-definition.module';
export * from './doughnut-chart/doughnut-chart-definition';
// export * from './doughnut-chart/data-validator';
export * from './doughnut-chart/doughnut-chart.component';
export * from './header/header-definition.module';
export * from './header/header-definition';
