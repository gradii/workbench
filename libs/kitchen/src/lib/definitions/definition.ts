import { ElementRef, InjectionToken, Provider, Type, ViewRef } from '@angular/core';
import { KitchenComponent, SyncReasonMsg } from '@common/public-api';
import { Observable } from 'rxjs';

import { Slot } from './definition-utils';
import { KitchenDataValidator } from '../workflow/data/data-validator.service';

export const COMPONENT_DEFINITION = new InjectionToken('Component Definition');
export const COMPONENT_META_DEFINITION = new InjectionToken('Component Meta Definition');

export const WIDGET_META_DEFINITION = new InjectionToken('Widget Meta Definition');

export type ComponentPropsMapper = (instance, props) => void;

// TODO get rid of these any
export interface Definition {
  id: string;
  componentType: Type<any>;
  definition: Type<any>;
  dataTrigger?: boolean;
  dataConsumer?: boolean;
  dataValidator?: KitchenDataValidator;
  propsMapper?: ComponentPropsMapper;

  factory(bindings?: any): KitchenComponent;

  inHeaderFactory?(bindings?: any): KitchenComponent;

  inSidebarFactory?(bindings?: any): KitchenComponent;
}

export interface MetaDefinition {
  definition: Definition;
  name: string;
  icon: string;
  order: number;
  headerSupport: boolean;
  sidebarSupport: boolean;
  locked?: boolean;
  tags: string[];
}

export interface WidgetMetaDefinition extends MetaDefinition {
  previewImage: string;
}

export function createDefinitionProvider(definition: Definition): Provider {
  return {
    provide: COMPONENT_DEFINITION,
    useValue: definition,
    multi: true,
  };
}

export function createMetaDefinitionProvider(metaDefinition: MetaDefinition): Provider {
  return {
    provide: COMPONENT_META_DEFINITION,
    useValue: metaDefinition,
    multi: true
  };
}

export function createWidgetMetaDefinitionProvider(metaDefinition: WidgetMetaDefinition): Provider {
  return {
    provide: WIDGET_META_DEFINITION,
    useValue: metaDefinition,
    multi: true
  };
}

export interface DefinitionContext<T> {
  view?: View<T>;
  devUIEnabled$?: Observable<boolean>;
  syncReason?: SyncReasonMsg;
  component?: KitchenComponent;
}

export class View<T> {
  instance: T;
  slots: {
    [key: string]: Slot;
  };
  viewRef?: ViewRef;
  element: ElementRef;
  properties$?: Observable<any>;
  elementChange$?: Observable<ElementRef>;
  draggable$?: Observable<boolean>;
  editable$?: Observable<boolean>;

  updateDynamicSlots?(): void;
}
