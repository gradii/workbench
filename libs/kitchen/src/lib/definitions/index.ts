export * from './definition'
export * from './definitions.module'

export * from './definition-utils'
export * from './view-definition-registry.service'
