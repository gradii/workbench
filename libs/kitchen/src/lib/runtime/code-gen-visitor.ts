import { KitchenComponent, KitchenSlot } from '@common/public-api';
import { PureElementNode, PureNode } from './builder/builder-node/pure-node';
import { TemplateBuilder } from './builder/template-builder';
import { mixinSpaceFeature } from './feature-mixins/mixin-space-feature';
import { mixinAccordion } from './mixins/mixin-accordion';
import { mixinButton } from './mixins/mixin-button';
import { mixinCard } from './mixins/mixin-card';
import { mixinCheckbox } from './mixins/mixin-checkbox';
import { mixinContentSlot } from './mixins/mixin-content-slot';
import { mixinHeading } from './mixins/mixin-heading';
import { mixinIcon } from './mixins/mixin-icon';
import { mixinSlot } from './mixins/mixin-slot';
import { mixinSpace } from './mixins/mixin-space';

export class CodeGenVisitor extends mixinContentSlot(
  mixinSlot(
    mixinIcon(
      mixinAccordion(
        mixinCheckbox(
          mixinButton(
            mixinHeading(
              mixinCard(
                mixinSpace(
                  mixinSpaceFeature(
                    class {
                    }
                  )
                )
              )
            )
          )
        )
      )
    )
  )
) {

  ident            = 0;
  output: string[] = [];

  public templateBuilder = new TemplateBuilder();

  elementNode: PureElementNode;

  visitElement(node: KitchenComponent | KitchenSlot): PureElementNode {
    if (!Object.hasOwnProperty.call(node, 'definitionId')) {
      return this.visitElementSlot(node as KitchenSlot);
    } else {
      if ((node as KitchenComponent).definitionId === 'space') {
        return this.visitElementSpace(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'space-feature') {
        return this.visitElementSpaceFeature(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'card') {
        return this.visitElementCard(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'button') {
        return this.visitElementButton(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'checkbox') {
        return this.visitElementCheckbox(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'accordion') {
        return this.visitElementAccordion(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'heading') {
        return this.visitElementHeading(node as KitchenComponent);
      } else if ((node as KitchenComponent).definitionId === 'icon') {
        return this.visitElementIcon(node as KitchenComponent);
      } else {
        throw new Error('unknown element');
      }
    }
  }

  incIdent(level: number = 1) {
    this.ident += level;
  }

  decIdent(level: number = 1) {
    this.ident -= level;
  }

  write(content: string) {
    this.output.push(content);
  }

  writeLn(content: string) {
    this.output.push(`${content}`);
  }

  identWrite(content: string) {
    this.output.push(`${'  '.repeat(this.ident)}${content}`);
  }

  identWriteLn(content: string) {
    this.output.push(`${'  '.repeat(this.ident)}${content}\n`);
  }

  incWrite(content: string, ident?: number) {
    this.incIdent(ident);
    this.identWrite(content);
  }

  decWrite(content: string, ident?: number) {
    this.decIdent(ident);
    this.identWrite(content);
  }

  writeDec(content: string, ident?: number) {
    this.identWrite(content);
    this.decIdent(ident);
  }

  getOutput() {
    return this.output.join('');
  }

  flush() {
    const output = this.getOutput();
    this.output  = [];
    return output;
  }

  getElementOutput(elementNodes: PureNode[]){
    return this.templateBuilder.build(elementNodes);
  }
}