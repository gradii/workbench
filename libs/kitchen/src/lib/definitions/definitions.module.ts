import { NgModule } from '@angular/core';
import { BakeryCommonModule } from '@common';
import {
  AccordionDefinitionModule, ButtonDefinitionModule, ButtonLinkDefinitionModule, CalendarDefinitionModule,
  CardDefinitionModule, CheckboxDefinitionModule, DatepickerDefinitionModule, DividerDefinitionModule,
  DoughnutChartDefinitionModule, HeaderDefinitionModule, HeadingDefinitionModule, IconDefinitionModule,
  IframeDefinitionModule, ImageDefinitionModule, InputDefinitionModule, LinkDefinitionModule, ListDefinitionModule,
  MapDefinitionModule, MenuDefinitionModule, MultipleAxisChartDefinitionModule, MultipleBarChartDefinitionModule,
  PieChartDefinitionModule, ProgressBarDefinitionModule, RadioDefinitionModule, SelectDefinitionModule,
  SidebarDefinitionModule, SidebarToggleDefinitionModule, SmartTableDefinitionModule, SpaceDefinitionModule,
  StepperDefinitionModule, TableDefinitionModule, TabsDefinitionModule, TextDefinitionModule
} from './components-definitions';
import { DefinitionUtilsModule } from './definition-utils';

import { DefinitionsComponent } from './definitions.component';
import {
  ContactsDefinitionModule, Login2DefinitionModule, Login3DefinitionModule, LoginDefinitionModule,
  MultiColumnFormDefinitionModule, OrdersDefinitionModule, ProfileDefinitionModule, Register2DefinitionModule,
  RegisterDefinitionModule, StatisticsDefinitionModule, StatusCardDefinitionModule, StatusChart2DefinitionModule,
  StatusChartDefinitionModule, TasksDefinitionModule, UsersDefinitionModule
} from './widget-definitions';

const DEFINITIONS = [DefinitionsComponent];

@NgModule({
  imports: [
    DefinitionUtilsModule,
    CardDefinitionModule,
    ButtonDefinitionModule,
    ButtonLinkDefinitionModule,
    InputDefinitionModule,
    TextDefinitionModule,
    SpaceDefinitionModule,
    CheckboxDefinitionModule,
    SelectDefinitionModule,
    RadioDefinitionModule,
    ProgressBarDefinitionModule,
    CalendarDefinitionModule,
    LinkDefinitionModule,
    TabsDefinitionModule,
    StepperDefinitionModule,
    PieChartDefinitionModule,
    MultipleAxisChartDefinitionModule,
    MultipleBarChartDefinitionModule,
    DoughnutChartDefinitionModule,
    MapDefinitionModule,
    TableDefinitionModule,
    SmartTableDefinitionModule,
    HeadingDefinitionModule,
    ImageDefinitionModule,
    MenuDefinitionModule,
    IconDefinitionModule,
    AccordionDefinitionModule,
    ListDefinitionModule,
    HeaderDefinitionModule,
    DatepickerDefinitionModule,
    SidebarDefinitionModule,
    SidebarToggleDefinitionModule,
    LoginDefinitionModule,
    RegisterDefinitionModule,
    ContactsDefinitionModule,
    UsersDefinitionModule,
    OrdersDefinitionModule,
    StatisticsDefinitionModule,
    TasksDefinitionModule,
    Login2DefinitionModule,
    Register2DefinitionModule,
    Login3DefinitionModule,
    ProfileDefinitionModule,
    StatusCardDefinitionModule,
    StatusChartDefinitionModule,
    StatusChart2DefinitionModule,
    MultiColumnFormDefinitionModule,
    IframeDefinitionModule,
    DividerDefinitionModule,

    BakeryCommonModule,
  ],
  exports: [...DEFINITIONS],
  declarations: [...DEFINITIONS]
})
export class DefinitionsModule {
}
