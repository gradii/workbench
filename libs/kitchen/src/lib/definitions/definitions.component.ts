import {
  AfterViewInit, ChangeDetectionStrategy, Component, ElementRef, Inject, Injectable, TemplateRef, ViewChild,
  ViewContainerRef
} from '@angular/core';
import { COMPONENT_DEFINITION, Definition } from './definition';

import { ViewDefinitionsService } from './view-definitions.service';

@Injectable()
export class StandardLibDefinitionService {
  constructor(
    @Inject(COMPONENT_DEFINITION) private definitions: Definition[],
    private viewDefinitionsService: ViewDefinitionsService
  ) {
  }

  defineStandardLib() {
    this.viewDefinitionsService.defineAll(this.definitions);
  }
}

@Component({
  selector       : 'puff-definitions',
  template       : '<ng-template></ng-template>',
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers      : [StandardLibDefinitionService]
})
export class DefinitionsComponent implements AfterViewInit {
  @ViewChild(TemplateRef, { read: ViewContainerRef }) host: ViewContainerRef;

  constructor(
    private _elementRef: ElementRef<HTMLElement>,
    private definitionsService: ViewDefinitionsService,
    private standardLibDefinitionsService: StandardLibDefinitionService
  ) {
  }

  ngAfterViewInit() {
    this.definitionsService.attach(this.host);
    this.standardLibDefinitionsService.defineStandardLib();

    // this._elementRef.nativeElement.remove();
  }
}
