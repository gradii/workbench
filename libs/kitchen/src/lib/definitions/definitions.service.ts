import { Inject, Injectable } from '@angular/core';

import { COMPONENT_DEFINITION, COMPONENT_META_DEFINITION, Definition, MetaDefinition } from './definition';

@Injectable({ providedIn: 'root' })
export class DefinitionsService {
  constructor(
    // @Inject(COMPONENT_DEFINITION) private definitions: Definition[],
    // @Inject(COMPONENT_META_DEFINITION) private metaDefinitions: MetaDefinition[]
  ) {
  }

  getDefinition(definitionId: string): Definition {
    return null
    // return this.definitions.find((d: Definition) => d.id === definitionId);
  }

  getMetaDefinition(definitionId: string): MetaDefinition {
    return null
    // return this.metaDefinitions.find((d: MetaDefinition) => d.definition.id === definitionId);
  }
}
