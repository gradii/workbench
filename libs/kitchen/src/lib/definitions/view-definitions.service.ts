import { ComponentFactoryResolver, ComponentRef, Injectable, ViewContainerRef } from '@angular/core';

import { Definition } from './definition';
import { ViewDefinitionRegistryService } from './view-definition-registry.service';
import { ComponentDefinition } from './definition-utils';

@Injectable({ providedIn: 'root' })
export class ViewDefinitionsService {
  private host: ViewContainerRef;

  constructor(
    private viewDefinitionRegistry: ViewDefinitionRegistryService,
  ) {
  }

  attach(host: ViewContainerRef) {
    this.host = host;
  }

  define(definition: Definition) {
    if (!this.host) {
      throw new Error('You need to attach DefinitionsService to the ViewContainerRef first.');
    }

    const definitionRef: ComponentRef<any> = this.renderDefinition(definition);
    this.defineComponent(definition, definitionRef);
  }

  defineAll(definitionList: Definition[]) {
    for (const definition of definitionList) {
      this.define(definition);
    }
  }

  private renderDefinition(definition: Definition): ComponentRef<any> {
    const definitionRef = this.host.createComponent(definition.definition);
    definitionRef.changeDetectorRef.detectChanges();

    return definitionRef;
  }

  private defineComponent(definition: Definition, definitionRef: ComponentRef<ComponentDefinition<any>>) {
    this.viewDefinitionRegistry.set(definition.id, definitionRef.instance.definition.template);
  }
}
