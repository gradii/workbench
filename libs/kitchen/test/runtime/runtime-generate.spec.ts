import { StateConverterService } from '@shared/communication/state-converter.service';
import { CodeGenVisitor } from '../../src/lib/runtime/code-gen-visitor';
import { Definitions } from './definitions';
import { Slots } from './slots';
import '@angular/compiler';
import { StateConverterTestingService } from '../../testing/state-converter-testing.service';

describe('runtime generate', () => {
  it('should generate a runtime', () => {
    const stateConverterService = new StateConverterTestingService();
    const data                  = stateConverterService.testConvertPage('1', Definitions, Slots);

    const visitor = new CodeGenVisitor();
    const slot    = data.content;
    visitor.visit(slot);

    const output = visitor.flush();

    expect(output).toMatchSnapshot();
  });
});
