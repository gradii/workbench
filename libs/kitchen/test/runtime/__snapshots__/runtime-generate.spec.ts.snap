// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime generate should generate a runtime 1`] = `
"    <div triDropFlexContainer uid=\\"Space\\" class=\\"space flex-column\\" triDropFlexContainerOrientation=\\"vertical\\" [triDropFlexContainerData]=\\"{id: '15786492785660.41803516438866173', name: 'Space', parentSlotId: '1'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
      <tri-card triDrag [triDragData]=\\"{id: 'FoflPrF9JT4OTg_RNErdT', name: 'Card3'}\\">
        <tri-card-header>
          <div triDropFlexContainer uid=\\"Space11\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: 'qjteET_qkTLh512JlMLGk', name: 'Space11', parentSlotId: 'cwDyXmOciWxyD0-JmWCxp'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
            <h6 uid=\\"Heading3\\" triDrag [triDragData]=\\"{id: 'UUdqcuIWMDbCG2wpxUk33', name: 'Heading3'}\\">
              Card Header
            </h6>
          </div>
        </tri-card-header>
        <tri-card-body>
          <div triDropFlexContainer uid=\\"Space12\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: 'LhJzILpx121ToMms8yU_I', name: 'Space12', parentSlotId: 'FxKLEFS_f-pk-U60Mro32'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
            <tri-icon uid=\\"Icon\\" triDrag triDrag [triDragData]=\\"{id: 'I7kn8fo3Exv9nHJzGuO5s', name: 'Icon'}\\" svgIcon=\\"fill:android\\">
            </tri-icon>
          </div>
        </tri-card-body>
        <tri-card-footer>
          <div triDropFlexContainer uid=\\"Space13\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: 'e_G8eg87o5qYyc--o0ec0', name: 'Space13', parentSlotId: 'Qm6H0N7R31W7sNHMoARi5'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
          </div>
        </tri-card-footer>
      </tri-card>
      <tri-card triDrag [triDragData]=\\"{id: 'Gb63MmCEbha-cNDiTjp8a', name: 'Card2'}\\">
        <tri-card-header>
          <div triDropFlexContainer uid=\\"Space8\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: '6cMyMyIaI32dsldfPPHI2', name: 'Space8', parentSlotId: '_G9bdhAQN-iUu1Zr5TWya'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
            <h6 uid=\\"Heading2\\" triDrag [triDragData]=\\"{id: '0i4zYZ_D2bFSxNlw3TnMK', name: 'Heading2'}\\">
              Card Header
            </h6>
          </div>
        </tri-card-header>
        <tri-card-body>
          <div triDropFlexContainer uid=\\"Space9\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: 'FFQDocQQajLsx2X2EuFw1', name: 'Space9', parentSlotId: 'S2QonCCC_-94t-aiuwBeB'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
            <button triButton uid=\\"Button\\" triDrag [triDragData]=\\"{id: 'rmFqmZy844Jpk8-6zUZz9', name: 'Button', parentSlotId: 'lAH6YoNK0QnzfZiifkIFS'}\\" color=\\"primary\\" [disabled]=\\"false\\" icon=\\"star\\" variant=\\"fill\\">Button</button>
          </div>
        </tri-card-body>
        <tri-card-footer>
          <div triDropFlexContainer uid=\\"Space10\\" class=\\"space flex-row\\" triDropFlexContainerOrientation=\\"horizontal\\" [triDropFlexContainerData]=\\"{id: '07zOrXvbNGK4w0o2WK1ne', name: 'Space10', parentSlotId: 'Oys8q1_-aQ-zIeihcuhiV'}\\" (triDropFlexContainerDropped)=\\"onDragDropped($event)\\">
            <button triButton uid=\\"Button1\\" triDrag [triDragData]=\\"{id: 'lgerPg736VKcIgI5ZXZvm', name: 'Button1', parentSlotId: '3crFqx2LrZ8Jtajd375yG'}\\" color=\\"danger\\" [disabled]=\\"false\\" icon=\\"star\\" variant=\\"fill\\">Button</button>
          </div>
        </tri-card-footer>
      </tri-card>
      <tri-accordion uid=\\"Accordion\\" triDrag>
      </tri-accordion>
    </div>"
`;
